File: Reflow/Tasks/ITask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Reflow.Tasks
{
    public delegate void TaskExecutionEventHandler(TaskExecutionContext context, TaskResult result);
    public interface ITask
    {
        string Name { get; set; }
        TaskResult Execute(TaskExecutionContext context);
        
        TaskExecutionEventHandler OnBeforeExecution { get; set; }
        TaskExecutionEventHandler OnAfterExecution  { get; set; }
    }
}

File: Java/src/main/java/com/nuarca/etl/tasks/ITask.java
package com.nuarca.etl.tasks;

public interface ITask {

    String getName() throws Exception ;

    void setName(String value) throws Exception ;

    TaskResult execute(TaskExecutionContext context) throws Exception ;

    TaskExecutionEventHandler getOnBeforeExecution() throws Exception ;

    void setOnBeforeExecution(TaskExecutionEventHandler value) throws Exception ;

    TaskExecutionEventHandler getOnAfterExecution() throws Exception ;

    void setOnAfterExecution(TaskExecutionEventHandler value) throws Exception ;

}
